src/compiler/checker.ts

```diff
@@ -2235,29 +2235,23 @@ namespace ts {
                         error(errorLocation, Diagnostics.Parameter_0_cannot_reference_identifier_1_declared_after_it, declarationNameToString(associatedDeclarationForContainingInitializerOrBindingName.name), declarationNameToString(errorLocation as Identifier));
                     }
                 }
-                if (result && errorLocation && meaning & SymbolFlags.Value && result.flags & SymbolFlags.Alias) {
-                    checkSymbolUsageInExpressionContext(result, name, errorLocation);
+                if (result && errorLocation && meaning & SymbolFlags.Value && result.flags & SymbolFlags.Alias && !(result.flags & SymbolFlags.Value) && !isValidTypeOnlyAliasUseSite(errorLocation)) {

```

Comment:
The only logical addition in this PR is `&& !(result.flags & SymbolFlags.Value)`. I decided to inline `checkSymbolUsageInExpressionContext` since it was only called once and it expected these symbol flag checks to hold true for the symbol passed in, which was very not obvious.